File: server/llm.py
# NOTES: Must install ollmaa from https://ollama.com/download
# Once downloaded, add to PATH and pull relevant ollama model --> ollama pull llama3.2:1b
# This file has a decent will take a few seconds to run
'''



DEPRECRATED. USE JS FILE INSTEAD


'''
from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:11434/v1",  # this endpoint is default for ollama, works for windows, not 100% for other OS
    api_key="ollama",
)

MODEL_NAME = "llama3.2:1b"

def get_similar_meals(meal: str) -> str:
    prompt = (
        f"Give me 3 meals that are similar to '{meal}'. "
        f"Consider ingredients, cuisine, flavor profile, and preparation style. "
        f"Return a clean bullet list only. No explanations."
    )

    response = client.chat.completions.create(
        model=MODEL_NAME,
        messages=[{"role": "user", "content": prompt}],
    )

    return response.choices[0].message.content

print(get_similar_meals(meal="chicken and rice"))



# from llm.py import get_similar_meals()
# get_similar_meals(meal="chicken and rice")

File: server/llm.js
// NOTES: Must install ollama from https://ollama.com/download
// Once downloaded, add to PATH and pull relevant ollama model --> ollama pull llama3.2:1b
// Heads up, this function will take a few seconds to run
import ollama from 'ollama';

const MODEL_NAME = 'llama3.2:1b';

export async function getSimilarMeals(meal) {
  if (!meal || typeof meal !== 'string') {
    throw new Error('Meal must be a non-empty string.');
  }

  const prompt = `
    Give me 5 meals that are similar to "${meal}".
    Consider ingredients, cuisine, flavor profile, and preparation style.
    Respond only with the meals' names, each on a new line.
    Do not add any text before or afterwards and do not write bullet points.
  `;

  const response = await ollama.chat({
    model: MODEL_NAME,
    messages: [
      { role: 'user', content: prompt }
    ]
  });

  return response.message.content.split('\n');
}